kore/src/Kore/Builtin/List.hs

```diff
@@ -254,11 +251,30 @@ evalGet =
                     (Builtin.appliedFunction . maybeBottom)
                         (Seq.lookup ix _list)
             emptyList <|> bothConcrete
-        )
       where
         maybeBottom =
             maybe Pattern.bottom Pattern.fromTermLike
 
+evalUpdate :: Builtin.Function
+evalUpdate =
+    Builtin.functionEvaluator evalUpdate0
+  where
+    evalUpdate0 :: Builtin.FunctionImplementation
+    evalUpdate0 resultSort [_list, _ix, value] =
+        Builtin.getAttemptedAxiom $ do
+            _list <- expectBuiltinList getKey _list
+            _ix <- fromInteger <$> Int.expectBuiltinInt getKey _ix
+            let len = Seq.length _list
+                ix
+                  | _ix < 0 =
+                    -- negative indices count from end of list
+                    _ix + len
```

Comment:
What if it wraps around twice? Should it be
```
_ix `mod` len
```
And I think you can remove the special case since this would work for both negative and positive numbers alike.